release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+import static java.lang.String.format;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * ReleaseNoteCreation creates the body of the release note for a repository of Scalar products. The
+ * body of the release note is written out as a markdown to standard output.
+ *
+ * <p>This script is assumed to be executed in a GitHub Actions workflow.
+ *
+ * <p>Note that it is needed that java 11 to execute this script since it is executed as a
+ * Single-File Source-code program.
+ */
+@SuppressWarnings("DefaultPackage")
+public class ReleaseNoteCreation {
+
+  private static final String DEBUG = System.getenv("DEBUG");
+
+  private final GitHubContext ghContext;
+
+  final Map<Category, List<ReleaseNoteText>> categoryMap = new EnumMap<>(Category.class);
+  final Map<String, List<ReleaseNoteText>> sameAsItems = new HashMap<>();
+
+  public static void main(String... args) throws Exception {
+    if (args.length != 4) {
+      System.err.printf(
+          "Usage:%n    java --source 11 %s.java <owner> <projectTitlePrefix> <version>"
+              + " <repository>%n%nExample:%n    java --source 11 %s.java scalar-labs ScalarDB 4.0.0"
+              + " scalardb%n",
+          ReleaseNoteCreation.class.getSimpleName(), ReleaseNoteCreation.class.getSimpleName());
+      System.exit(1);
+    }
+
+    String owner = args[0];
+    String projectTitlePrefix = args[1];
+    String version = args[2];
+    String repository = args[3];
+
+    ReleaseNoteCreation main =
+        new ReleaseNoteCreation(owner, projectTitlePrefix, version, repository);
+    main.createReleaseNote();
+  }
+
+  public ReleaseNoteCreation(
+      String owner, String projectTitlePrefix, String version, String repository) {
+    ghContext = new GitHubContext(owner, projectTitlePrefix, version, repository);
+  }
+
+  /** This constructor is only for test */
+  public ReleaseNoteCreation(GitHubContext ghContext) {
+    this.ghContext = ghContext;
+  }
+
+  public void createReleaseNote() throws Exception {
+    String projectId = ghContext.getProjectId();
+    List<String> prNumbers = ghContext.getPullRequestNumbers(projectId);
+
+    for (String prNumber : prNumbers) {
+      extractReleaseNoteInfo(prNumber);
+    }
+    assortSameAsItems();
+    outputReleaseNote();
+  }
+
+  void extractReleaseNoteInfo(String prNumber) throws Exception {
```

Comment:
I guess `prNumber` can be of a numeric type. Is there any reason for setting this as String?

Reply:
The `prNumber` is read via the `gh` command as a String and doesn't need to be the number in the script. That is the reason that I treated it as a String.